### PR TITLE
Export import

### PR DESCRIPTION
**Pull request by eliasreis54**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This PR introduce a new service to dojot.

* **What is the current behavior?** (You can also link to an open issue here)
Currently, the data cannot be exported and imported in others dojot.

* **What is the new behavior (if this is a feature change)?**
Whits this service, you will be export all data (templates, devices and flows), to other dojot.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
This PR is connected to dojot/dojot#769.

* **Other information**:

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const config = {
```
Comment (matheuscampanhaf) [inline src/config.js]: Have you considered importing configs from dojot-module-nodejs ?

File: src/utils/generateToken.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const generateToken = () => {
```
Comment (matheuscampanhaf) [inline src/utils/generateToken.js]: This is also done in dojot-module-nodejs

File: src/config.js (hunk 1 above)
Comment (eliasreis54) [reply src/config.js]: Nice! I'll do that!

File: src/utils/generateToken.js (hunk 2 above)
Comment (eliasreis54) [reply src/utils/generateToken.js]: Nice! I'll do that!

File: src/services/requestExport.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import { logger } from '@dojot/dojot-module-logger';
+import requestDevice from './requestsDevice';
```
Comment (raulnegreiros) [inline src/services/requestExport.js]: I particularly don't think these file names are meaningful. When you use it, like `requestDevice.requestDevice`, it is not clear what is happening, _maybe_ something like `DeviceManager.get`, `FlowManager.set` become more meaningful.

File: DockerFile [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+WORKDIR /opt/datamanager
+
+RUN apk --no-cache add gcc g++ bash musl-dev make python
+
+COPY package.json ./package.json
+
+RUN npm install
+
+RUN apk --no-cache add gcc g++ bash musl-dev make python
```
Comment (raulnegreiros) [inline DockerFile]: Who needs it and why you need to do it twice?

File: DockerFile [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+COPY . .
```
Comment (raulnegreiros) [inline DockerFile]: Particularly I prefer to add the src folder specifically, not the whole root directory

File: src/docs/apiary_.html [hunk 6]
```diff
@@ -0,0 +1,573 @@
...a href="#top" class="text-muted back-to-top"><i class="fa fa-toggle-up"></i>&nbsp;Back to top</a><div class="container"><div class="row"><nav><div class="resource-group"><div class="heading"><div class="chevron"><i class="open fa fa-angle-down"></i></div><a href="#">Resource Group</a></div><div class="collapse-content"><ul><li><a href="#import-post"><span class="badge post"><i class="fa fa-plus"></i></span>Configure a new templates, devices and flows</a></li><li><a href="#export-get"><span class="badge get"><i class="fa fa-arrow-down"></i></span>Get all data to be exported</a></li></ul></div></div></nav><div class="content"><header><h1 id="top">Data-manager</h1></header><section id="" class="resource-group"><h2 class="group-heading">Resource Group <a href="#" class="permalink">&para;</a></h2><div id="import" class="resource"><h3 class="resource-heading">Import <a href="#import" class="permalink">&nbsp;&para;</a></h3><div id="import-post" class="action post"><h4 class="action-heading"><div class="name">Configure a new templates, devices and flows</div><a href="#import-post" class="method post">POST</a><code class="uri">/import</code></h4><p>Returns all currently configured flows</p>
```
Comment (raulnegreiros) [inline src/docs/apiary_.html]: Do we really need to commit this file into the repository?

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+import bodyParser from 'body-parser';
+import Export from './routers/export';
+import Import from './routers/import';
+
+const app = express();
+
+app.use(bodyParser.json());
+app.use(Export);
+app.use(Import);
+app.listen(3000, () => {
```
Comment (raulnegreiros) [inline src/index.js]: would be nice if the port was configurable :D

File: src/routers/export.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+router.get('/export', (req, res) => {
+  requestExport()
+    .then((exportData) => {
+      res.json(exportData);
+    })
+    .catch((err) => {
+      const error = {
+        message: err.toString(),
+      };
+      res.status(400).json(error);
```
Comment (raulnegreiros) [inline src/routers/export.js]: This error is not described in the API

File: src/routers/import.js [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+  return requestImport.post(req.body)
+    .then((ret) => {
+      res.status(200).json(ret);
+    })
+    .catch((err) => {
+      const error = {
+        message: err.toString(),
+      };
+      res.status(400).json(error);
```
Comment (raulnegreiros) [inline src/routers/import.js]: This error is not described in the API

File: src/services/requestsImport.js [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+    });
+  });
+  return deviceList;
+}
+
+function changeIdDeviceFlow(newDevices, flows) {
+  const flowsList = flows;
+  flowsList.forEach((flow, indexDevice) => {
+    flow.flow.forEach((obj, index) => {
+      if ((obj.type === 'device in') || (obj.type === 'device out')) {
```
Comment (raulnegreiros) [inline src/services/requestsImport.js]: the actuate node Is missing, isn't?

File: DockerFile [hunk 11]
```diff
@@ -0,0 +1,15 @@
+FROM node:8-alpine
+
+WORKDIR /opt/datamanager
+
+RUN apk --no-cache add gcc g++ bash musl-dev make python
```
Comment (raulnegreiros) [inline DockerFile]: Just for curiosity, why we need gcc, python, etc?

File: src/config.js (hunk 1 above)
Comment (eliasreis54) [reply src/config.js]: Hey Matheus! Thank you for commenting. But, I was implementing dojot-module-nodejs, and I have an impression, the lib is too large for this application. With lib, we will have Kafka, for example, consumer and producer, and this is not necessary.

File: src/utils/generateToken.js (hunk 2 above)
Comment (eliasreis54) [reply src/utils/generateToken.js]: Hey Matheus! Thank you for commenting. But, I was implementing dojot-module-nodejs, and I have an impression, the lib is too large for this application. With lib, we will have Kafka, for example, consumer and producer, and this is not necessary.

File: src/docs/apiary_.html (hunk 6 above)
Comment (eliasreis54) [reply src/docs/apiary_.html]: No, it is not necessary. I will remote that.

File: src/index.js (hunk 7 above)
Comment (eliasreis54) [reply src/index.js]: I will provide it.

File: src/routers/export.js (hunk 8 above)
Comment (eliasreis54) [reply src/routers/export.js]: I will provide it.

File: src/routers/import.js (hunk 9 above)
Comment (eliasreis54) [reply src/routers/import.js]: I will provide it.

File: src/services/requestsImport.js (hunk 10 above)
Comment (eliasreis54) [reply src/services/requestsImport.js]: Yes, it is. I will provide it too.

File: DockerFile (hunk 11 above)
Comment (eliasreis54) [reply DockerFile]: Hmmm, good question... I made some tests, and it is not necessary to build an image. I will remove that.

**Pull request merged**